app/tasks_routes.py

```diff
@@ -0,0 +1,172 @@
+import datetime
+from app import db
+from app.models.task import Task
+from flask import Blueprint,jsonify,abort,make_response,request
+import requests
+from datetime import datetime
+import os
+from dotenv import load_dotenv
+
+tasks_bp = Blueprint('tasks_bp', __name__, url_prefix='/tasks')
+TASK_ID_PREFIX = '/<task_id>'
+
+def validate_task(task_id):
+    try:
+        task_id = int(task_id)
+    except:
+        abort(make_response({"message":f"Task {task_id} invalid"}, 400))
+
+    task = Task.query.get(task_id)
+
+    if not task:
+        abort(make_response({"message":f"Task {task_id} not found"}, 404))
+    
+    return task
```

Comment:
Nice helper function! Notice how `goal_routes` uses the exact same helper function? We can move this function into a new file and provide a `cls` parameter to make this function flexible enough to use on both `task` and `goal`. 

```python
def validate_task(cls, id):
    try:
        obj_id = int(id)
    except:
        abort(make_response({"message":f"{cls.__name__} {id} invalid"}, 400))

    obj = cls.query.get(id)

    if not obj:
        abort(make_response({"message":f"{cls.__name__}{id} not found"}, 404))
    
    return obj
```